extensions/config-yaml/runtime/src/test/java/io/quarkus/config/yaml/runtime/ApplicationYamlTest.java

```diff
@@ -21,19 +20,12 @@
  */
 public class ApplicationYamlTest {
 
-    static volatile SmallRyeConfig config;
-
     @BeforeAll
     public static void doBefore() {
         SmallRyeConfigBuilder builder = new SmallRyeConfigBuilder();
         builder.addDefaultSources().addDiscoveredConverters().addDiscoveredSources();
-        QuarkusConfigFactory.setConfig(config = builder.build());
-        Config conf = ConfigProvider.getConfig();
-        if (conf != config) {
-            ConfigProviderResolver cpr = ConfigProviderResolver.instance();
-            cpr.releaseConfig(conf);
-            ConfigProvider.getConfig();
-        }
+        SmallRyeConfig config = builder.build();
```

Comment:
This can be simplified. If we are getting a `SmallRyeConfig` instance from the builder and testing against it, we don't need to register it into the factory, because this is not even a Quarkus test.